analyser/analyser.go

```diff
@@ -0,0 +1,67 @@
+//Package analyser provides implementations for mobile app package analysis
+package analyser
+
+import (
+	"archive/zip"
+	"errors"
+	"log"
+	"regexp"
+	"strings"
+)
+
+const platformAndroid = "android"
+const PlatformIOS = "ios"
+
+type IAnalyser interface {
+	//getAppInfo returns populated AppInfo
+	getAppInfo() *AppInfo
+	//PrintAppInfo fetches and prints information about the mobile app package
+	PrintAppInfo()
+}
+
+//AppInfo is a struct, holding the recognisable information of the mobile app package
+type AppInfo struct {
+	AppId    string
+	Version  string
+	Build    string
+	IconPath string
+}
+
+//GetAnalyser returns the appropriate analyser implementation, based on the contents of the package
+func GetAnalyser(rc *zip.ReadCloser, originalFilePath *string) (IAnalyser, error) {
```

Comment:
it would be slightly better if this function would only receive the originalFilePath, as the zip read closer only used in this function, but originalFilePath is a required input for this function.